api/extension-guides/chat.md

```diff
@@ -0,0 +1,204 @@
+
+When a user explicitly mentions an `@agent` in their prompt, that prompt is forwarded to the extension that contributed that specific agent. Agents can respond using Markdown for simple text and image responses, or they can respond with a file tree or with buttons for a more interactive experience. For example, a file tree can be used as a preview when an agent is proposing to create a new workspace for the user. Agents can provide follow-ups for each response, imagine them as proposals on how to take the conversation further. To provide a smooth user experience, the whole API is streaming based. As already mentioned, agents can bring in sub commands - shortcuts to specific functionality. For example, the `@docker` agent might contribute a `/generate` sub command, resulting in the following example user prompt "`@docker /generate` a DOCKERFILE for workspace". The current syntax being explicit and concise can be a convenient time saver.
+
+![Chat concepts explanation](images/chat/chat.png)
+
+This screenshot shows the following chat concepts with a sample extension:
+1. `@cat` explicitly being called by the user via the `@` syntax
+2. Command explicitly called via the `/` syntax
+3. User query
+4. Copilot using the contributed `@cat` participant
+5. Markdown response that is fully controlled by the contributed `@cat` participant
+6. Code part of the markdown response
+7. [Button response](#response-stream)
+8. [Follow-ups](#follow-ups)
+9. Chat input for the user to continue. `description` of `@cat` is used as a placeholder
+
+
+## Chat Extension
+
+The [chat sample](https://github.com/microsoft/vscode-extension-samples/tree/main/chat-agent-sample) can be used as a starting point for your chat extension. It is a simple cat tutor that explains computer science topics using cat metaphors.
+
+![Diagram showing how extension can contribute to chat](images/chat/diagram.png)
+
+To start and register a chat participant you can do the following:
+
+```typescript
+const cat = vscode.chat.createChatParticipant('cat', handler);
+```
+
+> **Note:**: The Chat and Language Model API is in a [proposed state](https://code.visualstudio.com/api/advanced-topics/using-proposed-api) and we are actively working on adding more functionality. Share your feedback in [this GitHub issue](https://github.com/microsoft/vscode/issues/199908) or create new issues.
+
+### Commands
+
+Chat participants can contribute `/commands`, which are shortcuts to specific functionality provided by the extension. One of the tasks when answering questions is to determine the intent, understanding what you want to do. VS Code can infer that "Create a new workspace with Node.js Express Pug TypeScript" means that you want a new project, but "`@workspace /new` Node.js Express Pug TypeScript" is explicit, concise, and saves typing time. By pressing `/` VS Code will offer a list of registered commands with their description.
```

Comment:
> VS Code can infer that "Create a new workspace with Node.js Express Pug TypeScript" means that you want a new project

We don't yet do this kind of intent detection by default, as far as I know

Reply:
You are right. I changed "can" -> "could". 
Let me know if you still think we might be over-selling here.